File: 189-rotate-array/rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n=nums.size();
        vector<int>ans(n);
        k=k%n;
        for(int i=0;i<n;i++){
            ans[i]=nums[(n+i-k)%n];
        }
        for(int i=0;i<n;i++){
            nums[i]=ans[i];
        }
    }
};

File: 189-rotate-array/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        ArrayList<Integer> arr = new ArrayList<>();
        for(int num : nums) arr.add(num);

        k %= nums.length;
        Collections.rotate(arr,k);

        for(int i=0; i<nums.length; i++){
            nums[i]=arr.get(i);
        }
    }
}